plugins/database/quota/src/org/apache/cloudstack/api/response/QuotaResponseBuilderImpl.java

```
@@ -0,0 +1,419 @@
+
+        if (qb != null) {
+            throw new InvalidParameterValueException("Incorrect deposit date: " + adjustedStartDate + " there are balance entries after this date");
+        }
+
+        return addQuotaCredits(accountId, domainId, amount, updatedBy, adjustedStartDate);
+    }
+
+    @Override
+    public QuotaCreditsResponse addQuotaCredits(final Long accountId, final Long domainId, final Double amount, final Long updatedBy, final Date despositedOn) {
+        final short opendb = TransactionLegacy.currentTxn().getDatabaseId();
+        QuotaCreditsVO result = null;
+        TransactionLegacy txn = TransactionLegacy.open(TransactionLegacy.USAGE_DB);
+        try {
+            QuotaCreditsVO credits = new QuotaCreditsVO(accountId, domainId, new BigDecimal(amount), updatedBy);
+            s_logger.info("addQuotaCredits: Depositing " + amount + " on adjusted date " + despositedOn);
+            credits.setUpdatedOn(despositedOn);
+            result = _quotaCreditsDao.saveCredits(credits);
+        } finally {
+            txn.close();
+        }
+        TransactionLegacy.open(TransactionLegacy.CLOUD_DB).close();
+        final AccountVO account = _accountDao.findById(accountId);
+        final boolean lockAccountEnforcement = QuotaConfig.QuotaEnableEnforcement.value().equalsIgnoreCase("true");
+        final BigDecimal currentAccountBalance = _quotaBalanceDao.lastQuotaBalance(accountId, domainId, startOfNextDay(despositedOn));
+        if (lockAccountEnforcement && (currentAccountBalance.compareTo(new BigDecimal(0)) >= 0)) {
+            if (account.getState() == Account.State.locked) {
+                try {
+                    _regionMgr.enableAccount(account.getAccountName(), domainId, accountId);
+                    // _quotaMgr.sendQuotaAlert(account, currentAccountBalance, QuotaConfig.QuotaEmailTemplateTypes.QUOTA_UNLOCK_ACCOUNT);
+                } catch (Exception e) {
```

Catching ``Exception`` is broad -- catching all checked and unchecked exceptions.  Why are we catching all and ignoring all exceptions here?